client/src/components/FormBuilderModal/FormBuilderModal.js

```diff
@@ -1,104 +1,71 @@
-import React, { Component } from 'react';
-import i18n from 'i18n';
-import { Modal, ModalHeader } from 'reactstrap';
+import React, { useState } from 'react';
 import FormBuilderLoader from 'containers/FormBuilderLoader/FormBuilderLoader';
 import castStringToElement from 'lib/castStringToElement';
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
+import Modal from 'components/Modal/Modal';
+import pick from 'lodash/pick';
 
 const noop = () => null;
 
-class FormBuilderModal extends Component {
-  constructor(props) {
-    super(props);
-
-    this.handleSubmit = this.handleSubmit.bind(this);
-    this.handleHide = this.handleHide.bind(this);
-    this.clearResponse = this.clearResponse.bind(this);
-    this.handleLoadingError = this.handleLoadingError.bind(this);
-  }
-
-  /**
-   * Defines the form part of the Modal
-   *
-   * @returns {Component}
-   */
-  getForm() {
-    const { FormBuilderLoaderComponent } = this.props;
-    if (!this.props.schemaUrl) {
-      return null;
-    }
-    return (
-      <FormBuilderLoaderComponent
-        fieldHolder={{ className: classnames('modal-body', this.props.bodyClassName) }}
-        actionHolder={{ className: 'modal-footer' }}
-        autoFocus={this.props.autoFocus}
-        schemaUrl={this.props.schemaUrl}
-        onSubmit={this.handleSubmit}
-        onAction={this.props.onAction}
-        onLoadingError={this.handleLoadingError}
-        identifier={this.props.identifier}
-      />
-    );
-  }
-
-  /**
-   * Generates the response part of the Modal
-   *
-   * @returns {Component}
-   */
-  getResponse() {
-    if (!this.state || !this.state.response) {
-      return null;
-    }
-
-    let className = '';
-
-    if (this.state.error) {
-      className = this.props.responseClassBad;
-    } else {
-      className = this.props.responseClassGood;
-    }
-
-    return (
-      <div className={className}>
-        { castStringToElement('span', { html: this.state.response }) }
-      </div>
-    );
-  }
-
-  /**
-   * Removes the response from the state
-   */
-  clearResponse() {
-    this.setState({
-      response: null,
-    });
-  }
-
-  handleLoadingError(schema) {
-    const providesOnLoadingError = this.props.onLoadingError !== noop;
-    if (this.props.showErrorMessage || !providesOnLoadingError) {
-      const error = schema.errors && schema.errors[0];
-      this.setState({
-        response: error.value,
-        error: true,
-      });
+/**
+ * React component for displaying a Form in a Modal using a form schema URL
+ */
+function FormBuilderModal(props) {
+  const {
+    responseClassBad,
+    responseClassGood,
+    onLoadingError,
+    showErrorMessage,
+    onClosed,
+    onSubmit,
+    schemaUrl,
+    bodyClassName,
+    FormBuilderLoaderComponent,
+    children
+  } = props;
```

Comment:
```suggestion
const FormBuilderModal = ({
  bodyClassName,
  children,
  FormBuilderLoaderComponent,
  onLoadingError,
  onClosed,
  onSubmit,
  responseClassBad,
  responseClassGood,
  schemaUrl,
  showErrorMessage,
}) => {
```
While I certainly prefer just using the word `function`, it doesn't match the result of the codebase so use const + arrow instead.

Reply:
Done